### PR TITLE
ovl_Mir_Ray2 OK

### PR DESCRIPTION
**Pull request by ordlucas**

### CODE REVIEW

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.h [hunk 1]
```diff
@@ -5,9 +5,19 @@
...
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x7C];
+    /* 0x144 */ ColliderJntSph collider;
+    /* 0x164 */ ColliderJntSphElement elements;
+    /* 0x1A4 */ u16 unk1A4;
+    /* 0x1A6 */ s16 radius;
+    /* 0x1A8 */ f32 unk1A8;
+    /* 0x1AC */ LightNode* unk1AC;
+    /* 0x1B0 */ LightInfo info;
+    /* 0x1BE */ UNK_TYPE1 pad1BE[2];
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.h:20]: `light` and `lightInfo`, pad is struct packing and can be removed.

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c [hunk 2]
```diff
@@ -28,33 +30,106 @@ const ActorInit Mir_Ray2_InitVars = {
...
+    MirRay2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Mir_Ray2/MirRay2_Update.s")
+    if (this->unk1A4 & 1) {
+        if (Flags_GetSwitch(play, MIRRAY2_GET_SWITCH_FLAGS(thisx))) {
+            this->unk1A4 &= ~1;
+        }
+    } else {
+        func_80AF3FE0(this, play);
+        if ((this->actor.params & 0xF) != 1) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c:124]: Missed the macro here.

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c [hunk 3]
```diff
@@ -15,7 +15,9 @@ void MirRay2_Destroy(Actor* thisx, PlayState* play);
 void MirRay2_Update(Actor* thisx, PlayState* play);
 void MirRay2_Draw(Actor* thisx, PlayState* play);
 
-#if 0
+void func_80AF3F70(MirRay2* this);
+void func_80AF3FE0(MirRay2* this, PlayState* play);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c:19]: Neither of these forward declarations is needed, we don't put them here unless they are.

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c [hunk 4]
```diff
@@ -28,33 +30,106 @@ const ActorInit Mir_Ray2_InitVars = {
...
 
-#endif
+void func_80AF3F70(MirRay2* this) {
+    ColliderJntSphElement* elements;
+
+    this->collider.elements->dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->collider.elements->dim.worldSphere.center.y = this->actor.world.pos.y;
+    this->collider.elements->dim.worldSphere.center.z = this->actor.world.pos.z;
+    elements = this->collider.elements;
+    elements->dim.worldSphere.radius = this->unk1A8 * elements->dim.scale;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c:67]: Temp required?

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c [hunk 5]
```diff
@@ -28,33 +30,106 @@ const ActorInit Mir_Ray2_InitVars = {
...
+}
+
+void func_80AF3FE0(MirRay2* this, PlayState* play) {
+    if (this->actor.xzDistToPlayer < this->unk1A8) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c:71]: `range`, then, maybe.

File: src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c [hunk 6]
```diff
@@ -28,33 +30,106 @@ const ActorInit Mir_Ray2_InitVars = {
...
+    func_80AF3FE0(this, play);
+    this->unk1AC = LightContext_InsertLight(play, &play->lightCtx, &this->info);
+    Collider_InitJntSph(play, &this->collider);
+    Collider_SetJntSph(play, &this->collider, &this->actor, &sJntSphInit, &this->elements);
+    func_80AF3F70(this);
+    this->actor.shape.rot.x = 0;
+    this->actor.world.rot.x = this->actor.shape.rot.x;
+    if (MIRRAY2_GET_F(&this->actor) != 1) {
+        switchFlags = MIRRAY2_GET_SWITCH_FLAGS(&this->actor);
+        if ((switchFlags != 0x7F) && !Flags_GetSwitch(play, switchFlags)) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Mir_Ray2/z_mir_ray2.c:102]: This looks like another case where this temp isn't real.

**Review by jpburnett — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**